script/scaffold/templates/config_flow_oauth2/integration/api.py

```diff
@@ -35,7 +35,7 @@ def refresh_tokens(self) -> dict:
             self.session.async_ensure_token_valid(), self.hass.loop
         ).result()
 
-        return self.session.token
+        return self.session.token["access_token"]
```

Comment:
This is not a dictionary I think

Reply:
My reading of OAuth2Session (which matches the behavior seen when debugging a new integration with this scaffold) is that it is a dict with fields {"expires_at", "access_token", "refresh_token", "expires_in"}.  Basically returning the entire json response all the way up from  LocalOAuth2Implementation._token_request

https://github.com/home-assistant/core/blob/0cce35b23edffd7beffe6b817fa7a357265bf0ca/homeassistant/helpers/config_entry_oauth2_flow.py#L430

I could see an argument that maybe LocalOAuth2Implementation should change its behavior to actually return a "token" here.  Is there another similar oauth session implementation somewhere with a different behavior?

Reply:
What @sanyatuning probably means is that the `access_token` value inside `session.token` is not a dictionary.

Reply:
The idea here was to just return the last token request from the server, so you can use both the token type and the access token for your request.